File: backend/schemas/promo_code.py
from decimal import Decimal
from pydantic import BaseModel, ConfigDict

class promoCodeSchema(BaseModel):
    id: int
    code: str
    discount_perc: Decimal
    is_active: bool

    model_config = ConfigDict(from_attributes=True)

class PromoCodeCreate(BaseModel):
    code: str
    discount_perc: Decimal
    is_active: bool = True

    model_config = ConfigDict(from_attributes=True)

class PromoCodeUpdate(BaseModel):
    is_active: bool

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/promoCode.ts
export interface PromoCodeData {
  id: number;
  code: string;
  discount_perc: string;
  is_active: boolean;
}

export interface PromoCodeCreate {
  code: string;
  discount_perc: string;
  is_active?: boolean;
}

export interface PromoCodeUpdate {
  is_active: boolean;
}
